rust_src/src/process.rs

```diff
@@ -383,7 +383,7 @@ pub fn process_send_string(process: LispObject, mut string: LispStringRef) -> ()
     unsafe {
         send_process(
             cget_process(process),
-            string.data as *mut libc::c_char,
```

Comment:
In other places we have used `as_slice` for this kind of behavior. Should we also do that here?